torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -66,6 +60,9 @@ void propagateNode(Node* n) {
   for (size_t i = 0; i < outputs.size(); ++i) {
     try {
       auto new_output = graph->insertConstant(outputs[i]);
+      if (outputs[i].isNone()) {
```

Comment:
mentioned below:we should add an option TypePtr output_type to insertConstant, so it is clear what to do in cases where you are replacing an existing value with a constant and need to preserve its type.